src/main/java/org/example/daos/JobRoleDao.java

```diff
@@ -0,0 +1,58 @@
+package org.example.daos;
+
+import org.example.exception.DatabaseConnectionException;
+import org.example.models.JobRole;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+
+public class JobRoleDao {
+    public List<String> testConnection() throws SQLException,
+            DatabaseConnectionException {
+        List<String> databases = new ArrayList<>();
+
+        try (Connection connection = DatabaseConnector.getConnection()) {
+            Statement statement = connection.createStatement();
+
+            ResultSet resultSet = statement.executeQuery(
+                    "SHOW DATABASES;");
+
+            while (resultSet.next()) {
+                databases.add(resultSet.getString("Database"));
+            }
+        }
+
+        return databases;
+    }
+
+    public List<JobRole> getJobRoles() throws SQLException, DatabaseConnectionException {
```

Comment:
I would handle the SQL exception here in this method and only let one exception type bubble up

Reply:
My current implementation uses the SQL exception to determine if i should pass the status code 500 to in the response at the controller, i'm thinking another way to pass along this server error info would be to assign a meaning to a return value that specifies a server error?

Reply:
I ended up making a generic JobRoleException class which SQLException and DatabaseConnectionException extend which I then check which instance of exception in the controller try/catch to designate the correct response status, how is this implementation?